File: src/main/java/net/lifecity/mc/skillmaster/user/skillset/SkillSet.java
package net.lifecity.mc.skillmaster.user.skillset;

import lombok.Getter;
import net.lifecity.mc.skillmaster.skill.Skill;

import java.util.ArrayList;
import java.util.List;

/**
 * スキルキーが
 */
public class SkillSet extends ArrayList<SkillKey> {

    @Getter
    private final SkillButton button;

    public SkillSet(SkillButton button, Skill zero, Skill one, Skill two) {
        this.button = button;
        add(new SkillKey(button, 0, zero));
        add(new SkillKey(button, 1, one));
        add(new SkillKey(button, 2, two));
    }
}

File: src/main/java/net/lifecity/mc/skillmaster/user/skillset/SkillSet.kt
package net.lifecity.mc.skillmaster.user.skillset

data class SkillSet(
    val number: Int,// 0で現在のスキルセット
    var rightCard: SkillCard = SkillCard(SkillButton.RIGHT),
    var swapCard: SkillCard = SkillCard(SkillButton.SWAP),
    var dropCard: SkillCard = SkillCard(SkillButton.DROP)
) {
    val cards: List<SkillCard>
        get() = listOf(rightCard, swapCard, dropCard)

    val keyList: List<SkillKey>
        get() {
            val list = mutableListOf<SkillKey>()
            for (card in cards) {
                for (num in 0..2) {
                    list.add(card.skillKeySet.keyList[num])
                }
            }
            return list
        }
}